plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -955,8 +956,27 @@ private void finishOptimize(ConnectorSession session, IcebergTableExecuteHandle
                             .map(dataFile -> dataFile.path().toString())
                             .collect(toImmutableSet()));
         }
+        // The idea is that after optimize there should not be any delete files used anymore as
+        // the new data files have all these deletes already applied. This is possible
+        // due to the fact that we prevent skipping any data files in
+        // io.trino.plugin.iceberg.IcebergSplitSource.getNextBatch when delete files are present.
```

Comment:
What about deletion file shared between two data files?

one could have been filtered out based on optimize predicates (the WHERE clause)